src/plugin/modules/widgets/kbaseCatalogIndex.js

```diff
@@ -162,8 +162,7 @@ define([
 
         checkIsAdmin: function () {
             var self = this;
-            var me = self.runtime.service('session').getUsername();
-            return self.catalog.is_admin(me)
+            return self.catalog.is_admin()
```

Comment:
this changes the semantics of this call -- it should return the result of the is_admin call, although I can see in the rest of the code that the call is only used for side effect. But then again there is extra complexity in this module just to do the check for side effects of a single async call - it would be much more direct to just return the value and use it. Still, that isn't the fault of this change. So, to iterate, at least the value of the is_admin() call should be returned from the .then() method to retain the previous behavior.

Reply:
@eapearson I gave this more thought and because this would have to be coupled with the catalog changes all the way up it would be a huge hassle just like SearchAPI changes always were. Given that, I might just preserve the username parameter as before but ignore it except to throw an error if the username and token do not match. I think that should prevent misuse

What do you think?